src/timer.rs

```diff
@@ -0,0 +1,72 @@
+trait Timer {
+    fn start(milliseconds: u32) -> Self;
+    fn has_elapsed(self) -> Option<Self>
+    where
+        Self: Sized;
+}
+
+use libtock_core::syscalls;
+
+// see: https://github.com/tock/tock/blob/master/doc/syscalls/00000_alarm.md
+mod command_nr {
+    pub const _IS_DRIVER_AVAILABLE: usize = 0;
+    pub const GET_CLOCK_FREQUENCY: usize = 1;
+    pub const GET_CLOCK_VALUE: usize = 2;
+    pub const _STOP_ALARM: usize = 3;
+    pub const _SET_ALARM: usize = 4;
+}
+const DRIVER_NUMBER: usize = 0x00000;
+
+lazy_static! {
+    static ref CLOCK_FREQUENCY: usize =
+        syscalls::command(DRIVER_NUMBER, command_nr::GET_CLOCK_FREQUENCY, 0, 0).unwrap_or(0);
```

Comment:
I don't think we need this to be constant. We can do the syscall when creating a timer. That's already the case in the current implementation.

Reply:
Interesting, but isn't asking just once better? We have to assume that the frequency doesn't change over time anyway, right?

Reply:
Yes, I don't think we should plan for cases where the frequency would change over time. The tradeoff is between one syscall per timer creation (very small) and one extra dependency plus the associated code complexity of understanding when that "static syscall" is actually run. I don't think the "optimization" is worth it.

In particular, once this get in `Env`, the frequency would be stored in the environment, so no need for `lazy_static` and the optimization comes for free.

Reply:
Ah right, if we agree on that endgame, I'm happy with that!

Reply:
Removed lazy_static.